ARCs/arc-draft_action_routing.md

```diff
@@ -0,0 +1,67 @@
+---
+arc: <to be assigned>
+title: Application Action Routing
+description: Standardized handling of OnCompletion and ApplicationID actions 
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2024-01-02
+---
+
+## Abstract
+This ARC provides a standard way to handle an application call depending on the combination of the `OnCompletion` and `ApplicationID`.
+
+## Motivation
+Currently there is no way for someone to tell whether an app can be updated or deleted without doing a full static analysis on the TEAL.
+
+## Specification
+
+### TEAL
+The TEAL for an application **MUST** start with the following lines:
+
+```
+txn ApplicationID
+int 0
+>
```

Comment:
You could save an opcode by doing the following. But I think it would swap the "create_" and "call_" labels the other way round on the `switch`, if that's acceptable.

```suggestion
txn ApplicationID
!
```

Reply:
Nice! Changed in [cdfbdb7](https://github.com/algorandfoundation/ARCs/commit/cdfbdb7fb335667b24b20735ce498095eaaac4c1)